Mathlib/AlgebraicTopology/SimplicialObject/Coskeletal.lean

```diff
@@ -0,0 +1,116 @@
+class IsCoskeletal : Prop where
+  nonempty_isRightKanExtension :
+    Nonempty (IsRightKanExtension X (𝟙 ((Truncated.inclusion n).op ⋙ X)))
+
+variable [X.IsCoskeletal n]
+
+/-- If `X` is `n`-cosketal, then `𝟙 ((Truncated.inclusion n).op ⋙ X)` defines a right Kan
+extension of `(Truncated.inclusion.op ⋙ X)` along `(Truncated.inclusion n).op`. -/
+theorem IsCoskeletal.isRightKanExtension :
+    IsRightKanExtension X (𝟙 ((Truncated.inclusion n).op ⋙ X)) :=
+  IsCoskeletal.nonempty_isRightKanExtension.some
+
+/-- If `X` is `n`-coskeletal, then `rightExtensionInclusion X n` is a terminal object in the
+category `RightExtension (Truncated.inclusion n).op (Truncated.inclusion.op ⋙ X)`. -/
+noncomputable def IsCoskeletal.isUniversalOfIsRightKanExtension :
+    (rightExtensionInclusion X n).IsUniversal := by
+  have := isRightKanExtension X n
+  apply Functor.isUniversalOfIsRightKanExtension
+
+variable [∀ (F : (SimplexCategory.Truncated n)ᵒᵖ ⥤ C),
+    (SimplexCategory.Truncated.inclusion n).op.HasRightKanExtension F]
+
+/-- There is a map of costructured arrows from `rightExtensionInclusion X n` to the right extension
+of the `n`-truncation of `X` defined by the counit of `coskAdj n`. -/
+noncomputable def Truncated.rightExtensionCosk.hom : Truncated.rightExtensionInclusion X n ⟶
+    RightExtension.mk _
+      ((coskAdj n).counit.app ((Truncated.inclusion n).op ⋙ X)) :=
+  CostructuredArrow.homMk ((coskAdj n).unit.app X) ((coskAdj n).left_triangle_components X)
+
+instance Truncated.isRightKanExtensionCosk : IsRightKanExtension
+    ((Truncated.cosk n).obj ((inclusion n).op ⋙ X))
+    ((coskAdj n).counit.app ((inclusion n).op ⋙ X)) := by
+  unfold Truncated.cosk coskAdj
+  rw [ranAdjunction_counit]
+  exact Functor.instIsRightKanExtensionObjRanAppRanCounit _ _
+
+/-- The map `coskRightExtension.hom X` is a natural transformation between two right Kan extensions
+of the diagram `Truncated.inclusion.op ⋙ X` and thus is an isomorphism. -/
+instance IsCoskeletal.coskRightExtension.isIsoHom : IsIso (rightExtensionCosk.hom X n) :=
+  isIso_of_isTerminal (IsCoskeletal.isUniversalOfIsRightKanExtension X n)
+    (((Truncated.cosk n).obj
+      ((Truncated.inclusion n).op ⋙ X)).isUniversalOfIsRightKanExtension
+        ((coskAdj n).counit.app ((Truncated.inclusion n).op ⋙ X)))
+      (rightExtensionCosk.hom X n)
+
+/-- The map `coskRightExtension.hom X` is a natural transformation between two right Kan extensions
+of the diagram `Truncated.inclusion.op ⋙ X` and thus is an isomorphism. -/
+noncomputable def IsCoskeletal.coskRightExtension.homIso :
+    Truncated.rightExtensionInclusion X n ≅ RightExtension.mk _
+      ((coskAdj n).counit.app ((Truncated.inclusion n).op ⋙ X)) :=
+  asIso (rightExtensionCosk.hom X n)
+
+/-- The isomorphism `X ≅ (cosk n).obj X` that exists when `X` is coskeletal and the
+`n`-coskeleton functor exists.-/
+noncomputable def IsCoskeletal.isoCosk : X ≅ (cosk n).obj X :=
```

Comment:
I would prefer a name like `isoCoskOfIsCoskeletal`.

Reply:
No problem. I'll change it back. I was trying for greater uniformity but really have no idea what I'm doing..

Reply:
When we introduce a typeclass (here `IsCoskeletal`) for certain objects (here `SimplicialObject`), it is often convenient to put some definitions or lemmas in the `SimplicialObject` namespace rather than `IsCoskeletal`. One reason is that we can use dot notation, e.g. `X.isoCoskOfIsCoskeletal n`.

Reply:
(Sorry, my fault here, I think I made bad naming suggestions above.)